Added docs for v9 to v10 upgrade (#3953)
umbraco/UmbracoDocs: Fundamentals/Setup/Upgrading/version-specific.md

Follow the steps in the [general upgrade guide](general.md), then these additional instructions for the specific versions. (Remember that new config files are not mentioned because they are already covered in the general upgrade guide.)

## Version 9 to version 10

> The upgrade path between Umbraco 9 and Umbraco 10 and be done directly by updating the Nuget references. Additional to that you will have to remove a couple of files from your solution.

Reviewer: ```suggestion
The upgrade path between Umbraco 9 and Umbraco 10 can be done directly by updating the NuGet references. Additionally, you will have to remove a couple of files from your solution.
```

> The upgrade path between Umbraco 9 and Umbraco 10 and be done directly by updating the Nuget references. Additional to that you will have to remove a couple of files from your solution.

Reviewer 2: ```suggestion
The upgrade path between Umbraco 9 and Umbraco 10 can be done directly by updating your project NuGet references. Additionally, you will need to update `Program.cs` and remove some folders from your solution.
```

> We updated the `Program` class in the `Program.cs` file to the following out of box:

Reviewer: ```suggestion
We updated the `Program` class in the `Program.cs` file to the following out of the box:
```

```cs
 public class Program
    {
        public static void Main(string[] args)
            => CreateHostBuilder(args)
                .Build()
                .Run();

        public static IHostBuilder CreateHostBuilder(string[] args) =>
            Host.CreateDefaultBuilder(args)
                .ConfigureUmbracoDefaults()
                .ConfigureWebHostDefaults(webBuilder =>
                {
                    webBuilder.UseStaticWebAssets();
                    webBuilder.UseStartup<Startup>();
                });
    }
```

The calls to `ConfigureUmbracoDefaults` and `webBuilder.UseStaticWebAssets()` are new.

> Furthermore it is recommended to remove the following files and folders:

Reviewer: ```suggestion
Furthermore, it is recommended to remove the following files and folders:
```

- `/wwwroot/umbraco`
- `/umbraco/PartialViewMacros`
- `/umbraco/UmbracoBackOffice`
- `/umbraco/UmbracoInstall`
- `/umbraco/UmbracoWebsite`

> - `/umbraco/lang`

Reviewer 2: I think this should be `/umbraco/config/lang` @eshanrnh can you check?

Author: That's correct @Jeavon

All these files are now served directly from a Razor Class Library (Umbraco.Cms.StaticAssets) or as embedded resources.

:::note

> Please note to upgrade to Umbraco 10, you need your database to be at least on Umbraco 8.18.

Reviewer: ```suggestion
To upgrade to Umbraco 10, you need your database to be at least on Umbraco 8.18.
```
Since it's already in a note tag we need not write Please note again 🙂
